posthog/hogql/constants.py

```diff
@@ -673,6 +673,10 @@
     "maxIntersectionsPositionIf": 3,
 }
 ADD_TIMEZONE_TO_FUNCTIONS = ("now", "NOW", "toDateTime", "parseDateTime", "parseDateTimeBestEffort")
+# Functions where we use a -OrNull variant by default
+ADD_OR_NULL_DATETIME_FUNCTIONS = ("toDateTime", "parseDateTime", "parseDateTimeBestEffort")
```

Comment:
Perhaps I'm misunderstanding something, but if I read it correctly, `assumeNotNull` gives the column a default value (`0`, `""`, `false` ,etc) if it's `null`. Thus we're effectively using the `OrDefault` versions, not `OrNull` versions? 🤔 

Might even be better to say something like "will wrap with `assumeNotNull`" instead of "or null/default"?

Reply:
Hmm, naming things is hard - these are the functions where the HogQL parser adds the `OrNull` e.g. we use `parseDateTime64BestEffortOrNull` as a replacement for `toDateTime`.

For some reason this causes a type error inside `tumble`, even though it would handle nulls. Thus I'm wrapping it with an `assumeNotNull `, which gets rid of the type error.

Having read a bit more about `assumeNotNull`, it seems to have unpredictable behaviour - thus switching to `ifNull` and then merging.

```sql
SELECT tumble(toDateTimeOrNull('2019-01-01 00:00:00'), toIntervalDay('2'))
-- Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Nested type Tuple(DateTime, DateTime) cannot be inside Nullable type: While processing tumble(toDateTimeOrNull('2019-01-01 00:00:00'), toIntervalDay('2')). (ILLEGAL_TYPE_OF_ARGUMENT)

SELECT tumble(toDateTime('2019-01-01 00:00:00'), toIntervalDay('2'))
-- ('2018-12-31 00:00:00','2019-01-02 00:00:00')

SELECT tumble(null, toIntervalDay('2'))
-- null
```